database/mariadb.go

```diff
@@ -0,0 +1,121 @@
+package database
+
+import (
+	"fmt"
+	"strings"
+	"os"
+	"path"
+
+	"github.com/gobackup/gobackup/helper"
+	"github.com/gobackup/gobackup/logger"
+)
+
+// Mariadb database
+//
+// type: mariadb
+// host: 127.0.0.1
+// port: 3306
+// socket:
+// databases:
+// username: root
+// password:
+// args:
+
+type MariaDB struct {
+	Base
+	host          		string
+	port          		string
+	socket        		string
+	databases     		[]string
+	username      		string
+	password      		string
+	args          		string
+}
+
+func (db *MariaDB) init() (err error) {
+	viper := db.viper
+	viper.SetDefault("host", "127.0.0.1")
+	viper.SetDefault("username", "root")
+	viper.SetDefault("port", 3306)
+
+	db.host = viper.GetString("host")
+	db.port = viper.GetString("port")
+	db.socket = viper.GetString("socket")
+	db.databases = viper.GetStringSlice("databases")
+	db.username = viper.GetString("username")
+	db.password = viper.GetString("password")
+
+	if len(viper.GetString("args")) > 0 {
+		db.args = viper.GetString("args")
+	}
+
+	// socket
+	if len(db.socket) != 0 {
+		db.host = ""
+		db.port = ""
+	}
+
+	return nil
+}
+
+func createdatabasesfile(databasesfile string, databases []string) error {
+	file, err := os.Create(databasesfile)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	for _, database := range databases {
+		_, err := file.WriteString(database + "\n")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (db *MariaDB) build() string {
+	dumpArgs := []string{}
+	if len(db.host) > 0 {
+		dumpArgs = append(dumpArgs, "--host", db.host)
+	}
+	if len(db.port) > 0 {
+		dumpArgs = append(dumpArgs, "--port", db.port)
+	}
+	if len(db.socket) > 0 {
+		dumpArgs = append(dumpArgs, "--socket", db.socket)
+	}
+	if len(db.username) > 0 {
+		dumpArgs = append(dumpArgs, "-u", db.username)
+	}
+	if len(db.password) > 0 {
+		dumpArgs = append(dumpArgs, `-p`+db.password)
+	}
+
+	if len(db.databases) > 0 {
+		databasesfile := path.Join(db.dumpPath, "databases-file.txt")
```

Comment:
Why is `database-file.txt`

Reply:
This need to avoid create a tmp file on system, try use `echo` or other `bash` way.

Reply:
MariaDB backup require syntactic with spaces between databases:
--databases Defines the databases and tables you want to back up.

--databases="database[.table] [ database[.table] ...]"
When I try to pass it to gobackup it break the line by spaces and I got:
--databases="database[.table]
[ database[.table] ...]"

I will fix this by passing only one database to back up.

Reply:
I changed the backup logic to single database. So file is not required anymore.